sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSinkSuite.scala

```diff
@@ -473,6 +476,73 @@ abstract class FileStreamSinkSuite extends StreamTest {
       assert(outputFiles.toList.isEmpty, "Incomplete files should be cleaned up.")
     }
   }
+
+  testQuietly("cleanup complete but invalid output for aborted job") {
+    withSQLConf(("spark.sql.streaming.commitProtocolClass",
+      classOf[PendingCommitFilesTrackingManifestFileCommitProtocol].getCanonicalName)) {
+      withTempDir { tempDir =>
+        val checkpointDir = new File(tempDir, "chk")
+        val outputDir = new File(tempDir, "output")
+        val inputData = MemoryStream[Int]
+        inputData.addData(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
+        val q = inputData.toDS()
+          .repartition(10)
+          .map { value =>
+            // we intend task failure after some tasks succeeds
+            if (value == 5) {
+              // put some delay to let other task commits before this task fails
+              Thread.sleep(100)
+              value / 0
+            } else {
+              value
+            }
+          }
+          .writeStream
+          .option("checkpointLocation", checkpointDir.getCanonicalPath)
+          .format("parquet")
+          .start(outputDir.getCanonicalPath)
+
+        intercept[StreamingQueryException] {
+          try {
+            q.processAllAvailable()
+          } finally {
+            q.stop()
+          }
+        }
+
+        import PendingCommitFilesTrackingManifestFileCommitProtocol._
+        val outputFiles = Files.walk(outputDir.toPath).iterator().asScala
+          .filter(_.toString.endsWith(".parquet"))
+        // there would be possible to have race condition:
+        // - some tasks complete while abortJob is being called
+        // we can't delete complete files for these tasks (it's OK since this is a best effort)
+        assert(!outputFiles.toList.exists(f => tracking.contains(f.toUri.getPath)),
+          "abortJob should clean up files reported as successful.")
+      }
+    }
+  }
+}
+
+object PendingCommitFilesTrackingManifestFileCommitProtocol {
```

Comment:
Maybe, you want `ManifestFileCommitProtocolSuite`.

Reply:
I took this approach since testing with `ManifestFileCommit` needs some bootstrapping works like mocking, might become a verbose implementation. That was just based on my expectation, so I'll revisit this and update here if it goes really verbose.

Reply:
Got it. Sure. No problem with the current location.

Reply:
Ah yes. I thought you suggested to add UTs to ManifestFileCommitProtocolSuite which only deals with ManifestFileCommitProtocol (unit tests instead of acceptance tests). If the matter was location and you're OK with leave this as it is, let's leave it. Thanks!